prebackup/prebackup_test.go

```diff
@@ -0,0 +1,68 @@
+package prebackup
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	k8upv1alpha1 "github.com/vshn/k8up/api/v1alpha1"
+	"github.com/vshn/k8up/job"
+)
+
+func TestPreBackup_getPodTemplates(t *testing.T) {
+	ns := "test-namespace"
+	backupObj := &k8upv1alpha1.Backup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-backup",
+			Namespace: ns,
+		},
+	}
+	tests := map[string]struct {
```

Comment:
This map could be defined outside of the function. Makes it re-usable and the function smaller and easier to read.

Reply:
It depends on the test. Not all test cases have the same input and output, thus a dynamic type with specialised fields within the function gives most flexibility. e.g. `expectPodTemplateNames` might not be needed in all tests

Reply:
Yes that's true, but moving it outside the function makes the code more readable IMHO.

Reply:
I tried to address your readability concern with a helper function, this should already help a lot.

If the test cases are defined outside, I would have to scroll up and down way more :) With table-like test cases, the test cases with different names, preconditions and expectations become more important than the test code, thus I prefer to have it together. If we can share the same cases with same preconditions and expectations for another test, I'll be happy to consolidate those outside, but in this single test I don't see much value.

Reply:
Okay let's move the discussions here: https://github.com/vshn/k8up/discussions/231 :)

Reply:
Fine by me. We likely have to touch every test anyway once agreed on something, this shouldn't be a blocker for this PR though.